subprojects/signing/src/main/java/org/gradle/plugins/signing/SigningExtension.java

```diff
@@ -305,7 +308,47 @@ public void execute(Sign task) {
         return result;
     }
 
-    private Sign createSignTaskFor(final CharSequence name, Action<Sign> taskConfiguration) {
+    /**
+     * Creates signing tasks that sign {@linkplain PublicationInternal#getPublishableArtifacts()} all publishable artifacts} of the given publications.
+     *
+     * <p>The created tasks will be named "sign<i>&lt;publication name capitalized&gt;</i>Publication".
+     * That is, given a publication with the name "mavenJava" the created task will be named "signMavenJavaPublication".
+     *
+     * The signature artifacts for the created tasks are added to the {@linkplain PublicationInternal#getPublishableArtifacts()} publishable artifacts} of the given publications.
+     *
+     * @param publications The publications whose artifacts are to be signed
+     * @return the created tasks.
+     * @since 4.8
+     */
+    @Incubating
+    public List<Sign> sign(Publication... publications) {
+        List<Sign> result = new ArrayList<Sign>(publications.length);
+        for (final Publication publication : publications) {
+            if (publication instanceof PublicationInternal) {
+                result.add(sign((PublicationInternal<?>) publication));
+            }
+        }
+        return result;
+    }
+
+    private <T extends PublicationArtifact> Sign sign(final PublicationInternal<T> publicationToSign) {
+        final Sign signTask = project.getTasks().create("sign" + capitalize((CharSequence) (publicationToSign.getName() + "Publication")), Sign.class, new Action<Sign>() {
+            public void execute(Sign task) {
+                task.sign(publicationToSign);
+            }
+        });
+        signTask.getSignatures().all(new Action<Signature>() {
+            public void execute(Signature signature) {
+                publicationToSign
+                    .addDerivedArtifact((T) signature.getSource(), signature.getFile())
```

Comment:
❌ The call to signature.getFile needs to be lazy, as the user can still change the signature type later. We should have test coverage for that scenario.

Reply:
Can we do this in a subsequent (your?) PR?

Reply:
Fine by me.